### PR TITLE
Update Canary Testing Documentation

### LINKED ISSUE #3758: Update Canary Testing Documentation
The Canary Testing documentation uses the deprecated preferred and required. https://github.com/googleforgames/agones/blob/65a997e5185afb1592797957a7bbcdf29226ec41/site/content/en/docs/Integration%20Patterns/canary-testing.md?plain=1#L34-L44 We should update the example to use selectors similar to the Game Server Allocation reference https://github.com/googleforgames/agones/blob/65a997e5185afb1592797957a7bbcdf29226ec41/site/content/en/docs/Reference/gameserverallocation.md?plain=1#L28-L40

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3758 

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/docs/Integration Patterns/canary-testing.md [hunk 1]
```diff
@@ -35,12 +35,17 @@ state that in our system, the label `canary: "true"` will be added to any canary
 apiVersion: "allocation.agones.dev/v1"
 kind: GameServerAllocation
 spec:
-  preferred:
-    - matchLabels:
-        canary: "true"
-  required:
-    matchLabels:
+  - matchLabels:
```
Comment (igooch) [inline site/content/en/docs/Integration Patterns/canary-testing.md:38]: @markmandel do you have any preference on if this examples uses the labels `canary: "true"` `canary: "false"` or `agones.dev/fleet: green-fleet` `agones.dev/fleet: blue-fleet`?

```
   selectors: 
     - matchLabels:
         canary: "true"
         agones.dev/fleet: green-fleet
```

File: site/content/en/docs/Integration Patterns/canary-testing.md [hunk 2]
```diff
@@ -35,12 +35,17 @@ state that in our system, the label `canary: "true"` will be added to any canary
...
+      canary: "true"
       agones.dev/fleet: stable
+    players:
```
Comment (igooch) [inline site/content/en/docs/Integration Patterns/canary-testing.md:41]: We can remove the "players" selector. (Apologies I realize it's probably a bit confusing from the original issue.)

**Review by igooch — COMMENTED** [review]
Missing the "selectors" field, and we should remove the players selector. Also included a quick question for Mark on how he wants the labels to look.

File: site/content/en/docs/Integration Patterns/canary-testing.md (hunk 2 above)
Comment (Kalaiselvi84) [reply site/content/en/docs/Integration Patterns/canary-testing.md]: I have removed the `players` selector.

File: site/content/en/docs/Integration Patterns/canary-testing.md [hunk 3]
```diff
@@ -32,15 +32,13 @@ To ensure we don't have to change the Allocation system every time we have a can
...
 kind: GameServerAllocation
 spec:
-  preferred:
+  selectors:
     - matchLabels:
         canary: "true"
-  required:
-    matchLabels:
-      agones.dev/fleet: stable
+        agones.dev/fleet: stable
```
Comment (markmandel) [inline site/content/en/docs/Integration Patterns/canary-testing.md:41]: So right now this is a "canary and fleet is 'stable'" - what we actually want is "check the canary label'ed GameServers, and if there are none left, then go to the 'stable' fleet.

So we want this instead:

```yaml
apiVersion: allocation.agones.dev/v1
kind: GameServerAllocation
spec:
  selectors:
    - matchLabels:
        canary: "true"
    - matchLabels:
        agones.dev/fleet: stable
```

Which will first look for the first selector, and if it fails go to the second.

Comment (Kalaiselvi84) [reply site/content/en/docs/Integration Patterns/canary-testing.md]: I've made the changes you recommended.

**Review by igooch — APPROVED** [review]
LGTM

**Pull request merged**